### ISSUE TITLE
[BUG] - <TestCase 22 - 1476>

### ISSUE THREAD

**Issue by Dexless** [LeetCode-Feedback/LeetCode-Feedback#22363]
### LeetCode Username

Dexless

### Problem Number, Title, and Link

1476 / Subrectangle Queries  / https://leetcode.com/problems/subrectangle-queries/description/

### Bug Category

Incorrect test case *(Output of test case is incorrect as per the problem statement)*

### Bug Description

The problem is wordy but essentially, the problem asks the user to write a solution that overwrites the content with a given value of a NxM matrix where N or M is greater than one another making the matrix a rectangle. Test case 21 asks for the user to override the matrix: [3,9,4],[5,6,10] 1,1 -> 5 creating the matrix [3,9,4],[5,5,10]. This is okay, the test case fails when the user is asked to overwrite 0,0 to 1,0 with the value 6 creating the matrix: [6,6,6],[6,5,10]. The test case goes on to retrieve the value of 0,1 which should be 6 because of the aforementioned overwrite. The test incorrectly believes the value should be 9. 

TLDR: A segment of a matrix is overwritten sequentially(Left->Right & Top->Bottom) and when the test case tries to retrieve the value which is between these origin and the end coordinates it gives an output of the default construction of the matrix.

### Language Used for Code

C++

### Code used for Submit/Run operation

```bash
//1476 / Subrectangle Queries  / https://leetcode.com/problems/subrectangle-queries/description/
// Created by Dexless (March 21, 2024)
// Testcase 22 Bug Report
class SubrectangleQueries {
    vector<vector<int>> r;
public:
    SubrectangleQueries(vector<vector<int>>& rectangle) {
        r = rectangle;
    }
    
    void updateSubrectangle(int row1, int col1, int row2, int col2, int newValue) {
        if(row1 - row2 + col1 - col2 == 0){
            r[row1][col1] = newValue;
        }

        for(int i = row1; i <= row2; i++){
            int rowSize = ((i - row2) != 0) ? r[row1].size() : col2;
            for(int j = col1; j < rowSize; j++){
                r[i][j] = newValue;
            }
        }
        r[row2][col2] = newValue;

        for (size_t i = 0; i < r.size(); ++i) {
        for (size_t j = 0; j < r[i].size(); ++j) {
            std::cout << r[i][j] << " ";
        }
        std::cout << std::endl;
        }
        cout << endl;
    }
    
    int getValue(int row, int col) {
        return r[row][col];
    }
};
```

### Expected behavior

The value that should be produced should be the value that is fed through the updateSubrectangle() function. In this case the value should be 6.

### Screenshots

![image_2024-05-21_001906193](https://github.com/LeetCode-Feedback/LeetCode-Feedback/assets/87399166/6140ffdf-5255-4132-a628-2fd0792831e4)

### Additional context

I'm fairly new to leetcode so I hope my code isn't too hard to read or anything. Lines 21-27 is for debugging and not essential to the algorithm. Removing it should not have any positive or negative side effects.
